src/zimcheck.cpp

```diff
@@ -398,24 +418,32 @@ void test_articles(const zim::File& f, ErrorLogger& reporter, ProgressBar progre
             baseUrl.resize( pos==baseUrl.npos ? 0 : pos );
 
             auto links = getLinks(it->getData());
+            // remove duplicates
+            sort(links.begin(), links.end());
+            links.erase(unique(links.begin(), links.end()), links.end());
             for(auto olink: links)
             {
                 if (olink.front() == '#')
                     continue;
                 if (isInternalUrl(olink)) {
+                    if (isOutofBounds(olink, baseUrl)) {
+                        std::ostringstream ss;
+                        ss << olink << " is out of bounds. Article: " << it->getLongUrl();
+                        reporter.addError(URL_INTERNAL, ss.str());
+                        continue;
+                    }
                     auto link = normalize_link(olink, baseUrl);
                     char nm = link[0];
                     std::string shortUrl(link.substr(2));
                     auto a = f.getArticle(nm, shortUrl);
                     if (!a.good())
                     {
                         int index = it->getIndex();
-                        if ((previousLink != link) && (previousIndex != index) )
+                        if (previousIndex != index)
                         {
                             std::ostringstream ss;
-                            ss << link << " (" << olink << ") was not found in article " << it->getLongUrl();
+                            ss << link << " (" << olink << ") was not found in article " << it->getLongUrl() << " | shortUrl: " << shortUrl << " ns: " << nm;
```

Comment:
The plan is to remove the notion of namespace.
It will be done in next API but while then, we tend to not be explicit about the namespace and use the "longUrl" as the url.

I don't think we should change the error message here.

Reply:
I reverted the log msg back to what it was before